lnwire/dyn_propose.go

```diff
@@ -7,6 +7,7 @@ import (
 	"github.com/btcsuite/btcd/btcec/v2"
 	"github.com/btcsuite/btcd/btcutil"
 	"github.com/lightningnetwork/lnd/fn"
+	"github.com/lightningnetwork/lnd/keychain"
```

Comment:
IMO we don't want to add this dep. It'll invariably introduce an import cycle: lnwire -> keychain -> lnwallet -> lnwire. Also we don't need to prescribe what interface or how something is signed at this level. `lnwire` is a low level serialization package, and doesn't typically hold biz logic.